### PR TITLE
Add layer4 blueprints to support building from ART rpms

### PR DESCRIPTION
**Pull request by jogeo**
This PR addresses [USHIFT-2344](https://issues.redhat.com//browse/USHIFT-2344) and provides a layer of blueprints intended to be used with RPMs build by ART.

### CODE REVIEW

File: test/bin/build_images.sh [hunk 1]
```diff
@@ -545,6 +545,10 @@ build_images.sh [-iIsdf] [-l layer-dir | -g group-dir] [-t template]
 
   -d      Dry run by skipping the composer start commands.
 
+  -e      Extract container images.
+
+  -E      Do not extract container images.
+
```
Comment (ggiguash) [inline test/bin/build_images.sh:551]: We should not need both options. Let's keep the extraction ON by default and add "do not extract" flag.

File: test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml
Comment (ggiguash) [inline test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml:1]: Is `microshift-release` a name we should use if the packages are coming from ART?

File: test/image-blueprints/layer4-art/group1/rhel.toml
[the remark this answers is not in the document]
Comment (jogeo) [reply test/image-blueprints/layer4-art/group1/rhel.toml:1]: layer1 requires the microshift-test-agent package. That rpm package is built from source.
Layer4 will be run without building the source but rather after downloading rpms from brew or the mirror.

File: test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name = "microshift_release"
+description = "A RHEL 9.2 image with the RPMs built from ART."
+version = "0.0.1"
+modules = []
+groups = []
+distro = "rhel-92"
+
+[[packages]]
+name = "microshift"
+version = "{{ .Env.SOURCE_VERSION }}"
```
[the remark this answers is not in the document]
Comment (jogeo) [reply test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml:10]: Right, the RPMs are first downloaded, then just run build_images.sh.
Build_images.sh already extracts the version from the microshift-release-info rpm.

SOURCE_VERSION=$(rpm -q --queryformat '%{version}' "${release_info_rpm}")

[the remark this answers is not in the document]
Comment (dhellmann) [reply test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml:10]: I guess if you download the RPMs and put them in the same location as the ones built from source, they work with all of the other existing images from other layers, too? That seems like a reasonable benefit to trade off against the fact that we couldn't test source RPMs and ART built RPMs together in that setup.

File: test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml
Comment (jogeo) [reply test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml:1]: I can pick a different name. The main goal here is to create a series of commits that have a common parent and ref so we can upgrade through the entire lineage; base-no-microshift --> rhel9.2 --> rhel9.3. The way the build_images.sh script works today is that it sets the ref to the value of name defined in the toml file. The result is that when you run 'rpm-ostree upgrade' matching refs are not found and it fails. The RF CI tests are not using 'rpm-ostree upgrade' but rather 'rpm-ostree rebase' so they don't hit this issue.

Comment (jogeo) [reply test/image-blueprints/layer4-art/group2/rhel-92_microshift-release.toml:1]: Setting an alias might also work as well, so we can keep the names unique.

**Pull request closed without merging**